.github/workflows/test-batch-processing.yml

```diff
@@ -84,3 +84,10 @@ jobs:
           else
                sct_testing testing/batch_processing/test_batch_processing.py
           fi
+
+      - name: Upload folder summaries
+        if: always()
+        uses: actions/upload-artifact@v4
+        with:
+          name: Data Folder Summary (${{ matrix.os }})
+          path: "testing/sct_example_data.csv"
```

Comment:
Comparing the outputs inside `sct_example_data` based for [**macOS vs. Ubuntu**](https://www.diffchecker.com/DGZB2NNP/), we can see that discrepancies only begin to occur when we get to the registration steps.

This makes me think that the ANTs binaries are the source of issue:

- https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3194

And this makes me especially interested in trying out the following fix:

- https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/4376

It's possible that our custom build environments (e.g. ITK versions) differ between platforms, thus causing the discrepancy.

I'm going to try zipping up the official binaries, then comparing the results of the outputs using this PR.

Reply:
On the whole, the cross-platform registration discrepancies didn't really improve by updating ANTs to the officially built binaries. 

There is one thing that did improve, though: un-registered DTI metrics.

- `master`: https://www.diffchecker.com/DGZB2NNP/

![image](https://github.com/user-attachments/assets/b0b243aa-ec0b-4f9b-9f91-5f702793e34e)

- v2.5.4: https://www.diffchecker.com/sTT8bMDP/

![image](https://github.com/user-attachments/assets/54e7e3f4-c536-4bce-a341-e35b79b29e83)

That's only a small change, though, so upgrading the ANTs binaries wasn't exactly fruitful for solving #3194.

Still, I hope that the "folder summary" idea is still useful for tracking regressions from PRs?